torch/csrc/jit/script/init.cpp

```diff
@@ -167,17 +163,44 @@ struct VISIBILITY_HIDDEN ConstantPythonValue : public PythonValue {
       return createConstant(loc, m, at::CPU(at::kFloat).scalarTensor(py::cast<float>(self)));
     } else if(py::isinstance<py::bool_>(self)) {
       return createConstant(loc, m, at::CPU(at::kByte).scalarTensor(py::cast<bool>(self)));
+    } else if(THPDevice_Check(self.ptr())) {
+      auto device = (THPDevice*) self.ptr();
+      auto t = as_tensor({static_cast<int64_t>(device->device.type), device->device.deviceInt64()});
```

Comment:
Is this creating a tensor on some device? If so, seems like a strange thing for the AST builder to be doing.

Reply:
I don't think so. Instead it is serializing the (device_type, device_id) into  a tensor so it can be first-class in the interpreter.

Reply:
Oh I see. A bit goofy but I suppose it is fine for now.